### DISCUSSION TITLE
"Calibre metadata search" results sorting - issue/quirk/???

### DISCUSSION THREAD

**Discussion by Veyska** [koreader/koreader#11536] [Q&A] [answered]
Is it intended/expected that Calibre metadata search results are sorting authors by "FN LN" rather than "LN FN" (so, display vs sort), or that titles/series aren't ignoring leading "A/An/The"/etc (so, "A Title" getting sorted before "Book")?

I can fairly easily "fix" the series sorting behavior, and quite possibly will, but I'm reluctant to muck up author/title data just to "fix" the sorting in KOReader, especially since that would probably also mess up the Title/Authors lines in the Book Information pop-up.

If pointed in the right direction I can provide logs/screenshots/etc, but a couple potentially relevant data points from my set-up:
* I sync via USB only.
* I still find the occasional exception, but basically all of my books have happy little green checkmarks in their "Title/Author sort" boxes in Calibre.

(Apologies if I missed something relevant, but I searched both issues and discussions for "Calibre" and skimmed every page of results and also skimmed the first dozen or so pages of issues for "sort" and the closest hit I found was an old issue regarding series sorting and leading \#s.)

**Comment by NiLuJe** [comment, marked as answer]
Yup, nothing particular is being done about leading articles in titles (and doing it right would require an i18n aware implementation).

As far as authors are concerned, we simplify the metadata cache to save memory, and I suspect it was simply easier to keep the "display" `authors` key, instead of keeping what essentially amounts to a duplicate in the calibre `author_sort` key and/or the `author_sort_map` logic, and/or dealing with the slightly tricky inversion logic ourselves.

**Comment by Veyska** [comment]
Bah humbug.  I don't do enough reading on my phone for it to be more than annoying mostly on the principle of things currently, but if/when I get an e-ink tablet...  I love the reading UI but now that I've noticed the sorting behavior I'm not going to be able to *not* be distracted by it.  Oh well, maybe someone will magically have "fixed" things by the time I'm looking to replace my dedicated ereader device with an ereader app.  😅  Thanks for the clarification, even if it's not the answer I was hoping for.

(FWIW, if some awesome person sees this somewhen, the author sorting is by far the more annoying of the two for me and I'd imagine the simpler to solve.  😉)
